object_database/web/cells/cells.py

```diff
@@ -2870,6 +2877,28 @@ def onMessage(self, msgFrame):
                     self.currentIteration = msgFrame["iteration"]
 
                 self.selectionSlot.set(msgFrame["selection"])
+        elif msgFrame["event"] == "scrolling":
+            self.firstVisibleRow = msgFrame["firstVisibleRow"]
+            if self.onFirstRowChange is not None:
+                self.onFirstRowChange(self.firstVisibleRow)
+
+    def setFirstVisibleRow(self, rowNum):
+        """ Send a message to set the first visible row of the editor to
+        rowNum.
+
+        Parameters
+        ----------
+        rowNum : int
+        """
+        dataInfo = {"firstVisibleRow": rowNum}
+        # stage this piece of data to be sent when we recalculate
+        if self.exportData.get("dataInfo") is None:
+            self.exportData["dataInfo"] = [dataInfo]
+        else:
+            self.exportData["dataInfo"].append(dataInfo)
+
+        self.wasDataUpdated = True
+        self.markDirty()
```

Comment:
This is a great example of a place where it would be nice to be able to directly send a message to the component from the python cell, instead of forcing a recalc. Something like `self.messageJsComponent(dict(firstVisibleRow=rowNum))`